### PR TITLE
Improved handling of variable scopes for IDE0018 (inline declaration)

### LINKED ISSUE #18076: IDE0018 suggested incorrectly withing usings
**Version Used**: 
VisualStudio/15.0.0+26228.9

**Steps to Reproduce**:
```c#
class Example
{
    static void Main()
    {
        var fileDescriptors = GetFileDescriptors();
        string fileType;
        using (var f = GetFile(fileDescriptors, out fileType))
        {
            /* Do things */
        }
        Console.WriteLine(fileType);
    }
}
```

**Expected Behavior**:
IDE0018 is not suggested

**Actual Behavior**:
IDE0018 suggests:
```c#
class Example
{
    static void Main()
    {
        var fileDescriptors = GetFileDescriptors();
        using (var f = GetFile(fileDescriptors, out string fileType))
        {
            /* Do things */
        }
        Console.WriteLine(fileType);
    }
}
```
This causes error CS0103 `The name 'fileType' does not exist in the current context`

### PR DESCRIPTION
**Pull request by sharwell**
Fixes #18076

This pull request corrects the following errors and usability issues related to IDE0018:

* `using` statements are not being treated as a variable scope
* `fixed` statements are not being treated as a variable scope
* `try` statements are not being treated as a variable scope (specifically in the case of exception filters)
* Definite assignment analysis is disabled for variables without an initializer, resulting in suggestions getting reported which cause more problems than good

@dotnet/roslyn-ide for review

### CODE REVIEW

File: src/Features/CSharp/Portable/InlineDeclaration/CSharpInlineDeclarationDiagnosticAnalyzer.cs [hunk 1]
```diff
@@ -250,12 +250,6 @@ private void AnalyzeSyntaxNode(SyntaxNodeAnalysisContext context, INamedTypeSymb
             // In this case, inlining the 'i' would cause it to longer be definitely
             // assigned in the WriteLine invocation.
 
-            if (localDeclarator.Initializer == null)
```
Comment (sharwell) [inline src/Features/CSharp/Portable/InlineDeclaration/CSharpInlineDeclarationDiagnosticAnalyzer.cs]: :memo: I'm concerned about the performance implications of this change. I find that it's a clear usability win and the test case seems valid, but the solution results in a great deal of additional analysis. I'm not sure what the best approach to take here is.

File: src/EditorFeatures/CSharpTest/InlineDeclaration/CSharpInlineDeclarationTests.cs [hunk 2]
```diff
@@ -1309,6 +1309,138 @@ private static bool TryExtractTokenFromEmail(out string token)
...
+            await TestMissingAsync(
+@"
+using System;
+
+class C
+{
+    static void Main(string[] args)
+    {
+        string [|token|];
+        using (GetDisposableAndValue(out token))
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/CSharpTest/InlineDeclaration/CSharpInlineDeclarationTests.cs]: this concerns me.  from my recollection, out-vars in using scope should be in scope outside of hte out-var.

Comment (CyrusNajmabadi) [inline src/EditorFeatures/CSharpTest/InlineDeclaration/CSharpInlineDeclarationTests.cs]: Ok.  Discussed with mads.  The using behavior in the compiler is expected.

File: src/Features/CSharp/Portable/InlineDeclaration/CSharpInlineDeclarationDiagnosticAnalyzer.cs (hunk 1 above)
Comment (CyrusNajmabadi) [inline src/Features/CSharp/Portable/InlineDeclaration/CSharpInlineDeclarationDiagnosticAnalyzer.cs]: I'm not concerned.

**Comment by CyrusNajmabadi** [comment]
👍 Thanks for the tests.

**Pull request merged**